### PR TITLE
Appdata and translation related patches

### PR DESCRIPTION
**Pull request by sabriunal**
### appdata: Update appdata

- Add vcs-browser and translate URLs
- Add the translation tag
- Mark the developer name and release descriptions as untranslatable
- Remove redundant paragraph from the translation document

### po: Update Turkish translation

### CODE REVIEW

File: io.github.mightycreak.Diffuse.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ finish-args:
 modules:
   - name: diffuse
     builddir: true
+    run-tests: true
```
[the remark this answers is not in the document]
Comment (sabriunal) [reply io.github.mightycreak.Diffuse.yml:16]: It usefull for Develeopment tests that validates appdata, desktop and smilar files. I assumed the yml file was development purpose.

[the remark this answers is not in the document]
Comment (sabriunal) [reply io.github.mightycreak.Diffuse.yml:16]: That part: https://github.com/MightyCreak/diffuse/actions/runs/7434658424/job/20229039709?pr=220#step:5:196

**Comment by MightyCreak** [comment]
I just [fixed an issue with Python 3.12](https://github.com/MightyCreak/diffuse/pull/221) (which might be the reason why the CI/CD fails on your MR). Could you rebase your branch?

File: io.github.mightycreak.Diffuse.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MightyCreak) [reply io.github.mightycreak.Diffuse.yml:16]: Interesting!

It's indeed for dev purpose, but it's also used to generate the real manifest here: https://github.com/flathub/io.github.mightycreak.Diffuse/blob/master/io.github.mightycreak.Diffuse.yml

It's a manual process, so no worry. I'll update the release documentation: https://github.com/MightyCreak/diffuse/blob/main/docs/developers/release-process.md#create-new-release-on-flatpak

[the remark this answers is not in the document]
Comment (MightyCreak) [reply io.github.mightycreak.Diffuse.yml:16]: Wait, I've a better solution, I'll cherry-pick your commits in another MR and do the needed changes for everything not related to the translation. Gimme 15 mins :wink:

[the remark this answers is not in the document]
Comment (MightyCreak) [reply io.github.mightycreak.Diffuse.yml:16]: Can you check this PR, please? https://github.com/MightyCreak/diffuse/pull/222

**Comment by MightyCreak** [comment]
I merged #222 and rebased your branch on main. Apparently there is still an issue with the meson build. I think it's coming from the CI/CD configuration (maybe the appstreamcli doesn't exist or is too old). I'll check that.

**Comment by MightyCreak** [comment]
I found the issue!

I cherry-picked your changes and fixed the CI/CD here: #223

**Comment by MightyCreak** [comment]
I merged #223 and I rebased your branch on main. I think it should fix the CI/CD on your branch as well.

**Comment by MightyCreak** [comment]
Well... it still fails... 

I don't understand why... I tried the command line in an ubuntu:22.04 container and it worked. I think I'll have to find a way to see what's written in the CI/CD log file.

But it's getting late now, I'll tackle that tomorrow ;)

**Comment by MightyCreak** [comment]
Ok, I got the artifact and here's the error:

```text
W: io.github.mightycreak.Diffuse:50: url-invalid-type vcs-browser
   Invalid `type` property for this `url` tag. URLs of this type are not known in the AppStream
   specification.
```

**Comment by MightyCreak** [comment]
Well.. apparently I couldn't sleep until I found the source of the issue :sweat_smile: 

The issue was that `ubuntu-latest` is Ubuntu 22.04, which has appstream 0.15, which is too old to handle URL type `vcs-browser`. So I changed the docker image to ubuntu-23.04 (until `ubuntu-latest` will point to 23.04 too).

It's fixed now!
Merging your PR now!

Thanks!

**Pull request merged**